file_documentation/weekplanner/screens/weekplan_screen.md

```diff
@@ -1,2 +1,72 @@
 # weekplan_screen.dart
 This file documents the weekplan_screen.dart file in the weekplanner project.
+
+## Functionality
+The screen is the one responsible for showing all the days of the week. Each day in the weekplan_screen is displayed with all the planned activities for that single day. The colors of the days are a standard to help autistic children. From the Weekplan screen different functionality can be accessed:
+
+* Adding activies
+* Accessing an activity
+* Move mctivities
+
+The weekplan screen also uses the giraf_appbar which also adds functionality
+
+## Layout 
+The weekplan screen is displayed as seen here:
+![Layout of screen](../pictures/weekPlannerScreen.PNG) 
+The activity on Thursday is an activty marked as done, whereas the one on Wednesday is not done yet.
+
+## Buttons
+The buttons used in the weekplan screen are the addactivity buttons. One button is placed at the bottom of each day. 
+Tapping an addactivity button opens a new screen the pictogram_search_screen in which the user can search for a pictogram to add to that day. 
+
+
+## Code
+The weekplan_screen is a widget, thus it have to implement the build function, as seen here:
+```dart
+@override
+  Widget build(BuildContext context) {
+    return StreamBuilder<WeekplanMode>(
+        stream: authBloc.mode,
+        builder: (BuildContext context,
+            AsyncSnapshot<WeekplanMode> weekModeSnapshot) {
+          return Scaffold(
+            appBar: GirafAppBar(
+                title: 'Ugeplan',
+                appBarIcons: (weekModeSnapshot.data == WeekplanMode.guardian)
+                    ? <AppBarIcon>[
+                        AppBarIcon.changeToCitizen,
+                        AppBarIcon.settings,
+                        AppBarIcon.logout,
+                      ]
+                    : <AppBarIcon>[AppBarIcon.changeToGuardian]),
+            body: StreamBuilder<UserWeekModel>(
+              stream: weekplanBloc.userWeek,
+              initialData: null,
+              builder: (BuildContext context,
+                  AsyncSnapshot<UserWeekModel> snapshot) {
+                if (snapshot.hasData) {
+                  return _buildWeeks(snapshot.data.week, context);
+                } else {
+                  return const Center(
+                    child: CircularProgressIndicator(),
+                  );
+                }
+              },
+            ),
+          );
+        });
+  }
+```
+The build method returns a streambuilder which uses the authBloc.mode. authBloc.mode tells whether a guardian or citizen uses the app. The layout of the screen is dependant on this. Inside the Scaffold another streambuilder is used, the stream is set to weekplanBloc.userWeek. The stream emits the userweek to be shown. If there is data in the stream *_buildweeks* is called. Otherwise a CircularProgressIndicator is shown. 
```

Comment:
This paragraph is very confusing and unstructured and should be rewritten.

```suggestion

The build method returns a `StreamBuilder` which uses the `AuthBloc`'s `mode` stream. The `AuthBloc`'s `mode` stream tells whether a guardian or citizen uses the app. The layout of the screen is dependant on this. The `StreamBuilder` inside the `Scaffold` uses the stream of `WeekplanBloc.userWeek`. This stream emits the active `UserWeek`. We call *_buildWeeks* if there is data in the stream, and a `CircularProgressIndicator` is shown otherwise.
```